src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -27,8 +27,12 @@ protected CompilationModuleGroup(CompilerTypeSystemContext typeSystemContext, IC
             _rootProvider = rootProvider;
         }
 
-        public abstract bool IsTypeInCompilationGroup(TypeDesc type);
-        public abstract bool IsMethodInCompilationGroup(MethodDesc method);
+        public abstract bool IsInCompilationGroup(TypeDesc type);
+        public abstract bool IsInCompilationGroup(MethodDesc method);
+        public abstract bool ShouldShareAcrossModules(TypeDesc type);
```

Comment:
How expensive is putting things in a comdat section when we're doing single file compilation?

Seems like we're going through lengths to avoid putting comdat foldable things in a comdat section for single file compilation by farming everything out through these. What's the benefit?